### PR TITLE
feat(ui): Add react concurrent mode config feature

### PR DESCRIPTION
**Pull request by scttcper**
In react 18, we'd like to be able to toggle on/off react 18's concurrent mode without rolling the entire ui back to react 17. Adds an option to allow us to toggle it on and off and exposes it early enough in the application bootstrap.

### CODE REVIEW

File: src/sentry/options/defaults.py [hunk 1]
```diff
@@ -413,6 +413,14 @@
     flags=FLAG_ALLOW_EMPTY | FLAG_AUTOMATOR_MODIFIABLE,
 )
 
+# React concurrent renderer
+register(
+    "organizations:react-concurrent-renderer-enabled",
+    type=Bool,
+    default=False,
+    flags=FLAG_ALLOW_EMPTY | FLAG_PRIORITIZE_DISK | FLAG_AUTOMATOR_MODIFIABLE,
+)
```
Comment (evanpurkhiser) [inline src/sentry/options/defaults.py:422]: I think we need something in getsentry to have this tied to the feature flag right

File: src/sentry/web/client_config.py [hunk 2]
```diff
@@ -221,6 +221,8 @@ def enabled_features(self) -> Iterable[str]:
             "organizations:multi-region-selector", actor=self.user
         ):
             yield "organizations:multi-region-selector"
+        if features.has("organizations:react-concurrent-renderer", actor=self.user):
```
Comment (evanpurkhiser) [inline src/sentry/web/client_config.py:224]: I think you could just implement this using `options`

Comment (scttcper) [reply src/sentry/web/client_config.py:224]: i thought that is what i did here are they not the same thing

**Review by evanpurkhiser — APPROVED** [review]
Looks good to me

**Pull request merged**